### PR TITLE
add optimistic gas computation while estimating gas

### LINKED ISSUE #3652: improve gas estimation by calculating optimistic gas

### PR DESCRIPTION
**Pull request by tclemos**
Closes #3652.

### What does this PR do?

implement the optimistic gas computation while estimating gas

### CODE REVIEW

File: state/test/forkid_common/common.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func InitTestState(stateCfg state.Config) *state.State {
 		panic(err)
 	}
 
-	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "zkevm-prover")
+	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "localhost")
```
Comment (agnusmor) [inline state/test/forkid_common/common.go:56]: is that ok? maybe you have changed it for local test and you forgot to undo the change

Comment (tclemos) [reply state/test/forkid_common/common.go:56]: yes, it works for both, local and CI

**Review by agnusmor — APPROVED** [review]
